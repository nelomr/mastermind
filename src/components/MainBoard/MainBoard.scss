.main-board {
    width: calc(100% - 175px);
    padding: 40px 20px 40px 80px;
    margin: 40px 0;
    background-color: $light-color;
    border-radius: 15px;
    box-shadow: 0px 2px 10px -2px $grey-color;

    @include breakpoint($min-desktop) {
        margin: 0 40px;
    }
}

.controls-board--check {
    margin-bottom: 20px;
}