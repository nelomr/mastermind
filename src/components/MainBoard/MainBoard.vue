<template>
    <div class="main-board">
      <RowBoard 
        v-for="(row, index) in defaultConfig.max_guesses"
        :key="'row'+index"
        :indexRow="index"
        :current="currentGuess"
        :reset="reset"
      />
    </div>
</template>

<script>
import { mapState } from 'vuex';
import RowBoard from '@/components/RowBoard/RowBoard';

export default {
  name: 'MainBoard',
  props: ['reset'],
  components: {
    RowBoard
  },
  computed: {
    ...mapState({
      defaultConfig: state => state.defaultConfig,
      currentGuess: state => state.currentGuess
    })
  },
  methods: {
    checkRow(index) {
      return this.currentGuess == index;
    }
  }
}
</script>

<style lang="scss">
  @import "./MainBoard.scss";
</style>