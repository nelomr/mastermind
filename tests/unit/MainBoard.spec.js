import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import MainBoard from '@/components/MainBoard/MainBoard.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('MainBoard.vue', () => {
  let config = {
    'num_colors': 6,
    'num_slots': 4, 
    'max_guesses': 5
  };

  let store;
  let computed;

  beforeEach(() => {
    computed = {
      defaultConfig() {
          return config;
      }
    };
    store = new Vuex.Store({});
  });
  const build = () => {
    const wrapper = shallowMount(MainBoard, {
      computed,
      mocks: {$store: store}
    });

    return {
        wrapper
    };
  };

  it('renders the component no button list', () => {
    const {wrapper} = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
})
