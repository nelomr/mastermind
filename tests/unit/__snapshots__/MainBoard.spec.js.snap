// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBoard.vue renders the component no button list 1`] = `
<div class="main-board">
  <rowboard-stub indexrow="0"></rowboard-stub>
  <rowboard-stub indexrow="1"></rowboard-stub>
  <rowboard-stub indexrow="2"></rowboard-stub>
  <rowboard-stub indexrow="3"></rowboard-stub>
  <rowboard-stub indexrow="4"></rowboard-stub>
</div>
`;
